common/src/main/java/com/wynntils/core/managers/ManagerRegistry.java

```diff
@@ -57,105 +56,88 @@ public static void init() {
      * <p>
      * Do not use this if you don't know what you are doing. Instead, register the manager as a feature dependency.
      * */
-    private static void registerPersistentDependency(Class<? extends CoreManager> manager) {
+    private static void registerPersistentManager(Class<? extends CoreManager> manager) {
         PERSISTENT_CORE_MANAGERS.add(manager);
-        ENABLED_MANAGERS.add(manager);
 
         WynntilsMod.registerEventListener(manager);
 
-        tryInitManager(manager);
-    }
-
-    public static void addDependency(Feature dependant, Class<? extends Model> dependency) {
-        if (PERSISTENT_CORE_MANAGERS.contains(dependency)) {
-            throw new IllegalStateException("Tried to register a persistent manager.");
+        if (!tryInitManager(manager)) {
+            WynntilsMod.error("Failed to initialized peristent manager " + manager.getName());
         }
-
-        MODEL_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
-
-        MODEL_DEPENDENCIES.get(dependency).add(dependant);
-
-        updateManagerState(dependency);
     }
 
-    public static void addFunctionDependency(
-            DependantFunction<?> dependantFunction, Class<? extends Model> dependency) {
-        if (PERSISTENT_CORE_MANAGERS.contains(dependency)) {
+    public static boolean addDependency(ModelDependant dependant, Class<? extends Model> manager) {
+        if (PERSISTENT_CORE_MANAGERS.contains(manager)) {
             throw new IllegalStateException("Tried to register a persistent manager.");
         }
 
-        FUNCTION_MODEL_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+        List<ModelDependant> managerDependencies =
+                MODEL_DEPENDENCIES.computeIfAbsent(manager, (k) -> new ArrayList<>());
 
-        FUNCTION_MODEL_DEPENDENCIES.get(dependency).add(dependantFunction);
+        managerDependencies.add(dependant);
 
-        updateManagerState(dependency);
+        return tryInitManager(manager);
     }
 
-    public static void removeDependency(Feature dependant, Class<? extends Model> dependency) {
-        if (PERSISTENT_CORE_MANAGERS.contains(dependency)) {
+    public static boolean removeDependency(ModelDependant dependant, Class<? extends Model> manager) {
+        if (PERSISTENT_CORE_MANAGERS.contains(manager)) {
             throw new IllegalStateException("Tried to unregister a persistent manager.");
         }
 
-        MODEL_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
```

Comment:
What was wrong with putIfAbsent?

Reply:
`computeIfAbsent` removes the get call, and in my opinion makes the logic more sound